serdes/serde-common/src/main/java/io/apicurio/registry/serde/AbstractSchemaResolver.java

```diff
@@ -94,25 +90,14 @@ public void configure(Map<String, ?> configs, boolean isKey, SchemaParser<S> sch
         Object ais = config.getArtifactResolverStrategy();
         Utils.instantiate(ArtifactResolverStrategy.class, ais, this::setArtifactResolverStrategy);
 
-        long checkPeriod = 0;
-        Object cp = config.getCheckPeriodMs();
-        if (cp != null) {
-            long checkPeriodParam;
-            if (cp instanceof Number) {
-                checkPeriodParam = ((Number) cp).longValue();
-            } else if (cp instanceof String) {
-                checkPeriodParam = Long.parseLong((String) cp);
-            } else if (cp instanceof Duration) {
-                checkPeriodParam = ((Duration) cp).toMillis();
-            } else {
-                throw new IllegalArgumentException("Check period config param type unsupported (must be a Number, String, or Duration): " + cp);
-            }
-            if (checkPeriodParam < 0) {
-                throw new IllegalArgumentException("Check period must be non-negative: " + checkPeriodParam);
-            }
-            checkPeriod = checkPeriodParam;
-        }
-        globalIdCacheByArtifactReference = new CheckPeriodCache<>(checkPeriod);
+        schemaCache.configureLifetime(config.getCheckPeriod());
```

Comment:
maybe encapsulate this inside a configure method by creating a config object and pass that config to the schema cache?